scripts/compose.py

```diff
@@ -1676,14 +1670,18 @@ def start_handler(self):
             print("Starting stack services..\n")
 
             # always build if possible, should be quick for rebuilds
-            if any("build" in service for service in compose["services"].values()):
+            build_services = [name for name, service in compose["services"].items() if 'build' in service]
+            if build_services:
                 docker_compose_build = ["docker-compose", "-f", docker_compose_path.name, "build", "--pull"]
                 if args["force_build"]:
                     docker_compose_build.append("--no-cache")
                 if args["build_parallel"]:
                     docker_compose_build.append("--parallel")
-                subprocess.call(docker_compose_build)
+                subprocess.call(docker_compose_build + build_services)
 
+            # pull any images
+            image_services = [name for name, service in compose["services"].items() if 'image' in service]
+            subprocess.call(["docker-compose", "-f", docker_compose_path.name, "pull"] + image_services)
```

Comment:
Why separate the build and pull commands?  Won't build without specifying the list of services pull anything needed?

Reply:
`docker-compose build` doesn't do anything for image-only services. And `docker-compose up` doesn't have a `--pull` flag (https://github.com/docker/compose/issues/3574 if you want to laugh/cry/despair). So AFAIK, separating it up is the only way we can do this

Reply:
Aha!  And this is needed because previously we didn't have images from a registry that would change but now we do - this is great then, thanks for explanation.